genvm/vm.go

```diff
@@ -223,6 +220,19 @@ func (v *VM) Apply(lctx ApplyContext, txs []types.Transaction, blockRewards []ty
 	hasher := hash.New()
 	encoder := scale.NewEncoder(hasher)
 	total := 0
+
+	tx, err := v.db.TxImmediate(context.Background())
+	if err != nil {
+		return nil, nil, err
+	}
+	defer tx.Release()
+
+	for _, reward := range rewardsResult {
+		if err := rewards.Add(tx, &reward); err != nil {
+			return nil, nil, fmt.Errorf("%w: %s", core.ErrInternal, err.Error())
+		}
+	}
+
 	ss.IterateChanged(func(account *core.Account) bool {
```

Comment:
this change trigger a concern that probably exists before.

the fact that the `StagedCache` holds on to a db executor concerns me a little. here the code is calling `ss.IterateChanged` that execute sql stmts inside a db transaction, with the possibility that inside `ss.ItrateChanged` it could potentially use it's own `sql.Executor` to do something.

would it be ok to pass a `sql.Executor` to `StagedCache.Get` so that the staged cache doesn't need a member of `sql.Exectuor`?

Reply:
i passed interface to the cache